hudi-utilities/src/main/java/org/apache/hudi/utilities/HoodieMetadataTableValidator.java

```diff
@@ -1057,6 +1083,74 @@ private void validateRecordIndex(HoodieSparkEngineContext sparkEngineContext,
     }
   }
 
+  private void validateSecondaryIndex(HoodieSparkEngineContext engineContext, HoodieMetadataValidationContext metadataContext,
+                                      HoodieTableMetaClient metaClient) {
+    if (!metaClient.getTableConfig().isMetadataPartitionAvailable(MetadataPartitionType.SECONDARY_INDEX)) {
+      return;
+    }
+    Collection<HoodieIndexDefinition> indexDefinitions = metaClient.getIndexMetadata().get().getIndexDefinitions().values();
+    for (HoodieIndexDefinition indexDefinition : indexDefinitions) {
+      validateSecondaryIndex(engineContext, metadataContext, metaClient, indexDefinition);
+    }
```

Comment:
It looks like you are already validating multiple secondary indexes, so is the TODO added in `SecondaryIndexSupport` still valid?

Reply:
Yes, it is still valid. Since there is no test for it right now.